src/napi/node_api_value.c

```diff
@@ -55,3 +74,111 @@ napi_status napi_create_string_utf8(napi_env env, const char* str,
   *result = AS_NAPI_VALUE(jval);
   return napi_ok;
 }
+
+napi_status napi_get_array_length(napi_env env, napi_value value,
+                                  uint32_t* result) {
+  jerry_value_t jval = AS_JERRY_VALUE(value);
+  *result = jerry_get_array_length(jval);
+  return napi_ok;
+}
+
+napi_status napi_get_prototype(napi_env env, napi_value object,
+                               napi_value* result) {
+  jerry_value_t jval = AS_JERRY_VALUE(object);
+  jerry_value_t jval_proto = jerry_get_prototype(jval);
+  *result = AS_NAPI_VALUE(jval_proto);
+  return napi_ok;
+}
+
+napi_status napi_get_value_bool(napi_env env, napi_value value, bool* result) {
+  jerry_value_t jval = AS_JERRY_VALUE(value);
+  if (!jerry_value_is_boolean(jval))
+    return napi_boolean_expected;
+  *result = jerry_get_boolean_value(jval);
+  return napi_ok;
+}
+
+napi_status napi_get_boolean(napi_env env, bool value, napi_value* result) {
+  JERRYX_CREATE(jval, jerry_create_boolean(value));
+  *result = AS_NAPI_VALUE(jval);
+  return napi_ok;
+}
+
+napi_status napi_get_global(napi_env env, napi_value* result) {
+  JERRYX_CREATE(jval, jerry_get_global_object());
+  *result = AS_NAPI_VALUE(jval);
+  return napi_ok;
+}
+
+napi_status napi_get_null(napi_env env, napi_value* result) {
+  JERRYX_CREATE(jval, jerry_create_null());
+  *result = AS_NAPI_VALUE(jval);
+  return napi_ok;
+}
+
+napi_status napi_get_undefined(napi_env env, napi_value* result) {
+  JERRYX_CREATE(jval, jerry_create_undefined());
+  *result = AS_NAPI_VALUE(jval);
+  return napi_ok;
+}
+
+#define DEF_NAPI_COERCE_TO(type, alias)                             \
+  napi_status napi_coerce_to_##type(napi_env env, napi_value value, \
+                                    napi_value* result) {           \
+    jerry_value_t jval = AS_JERRY_VALUE(value);                     \
+    jerry_value_t jval_result = jerry_value_to_##alias(jval);       \
+    *result = AS_NAPI_VALUE(jval_result);                           \
+    return napi_ok;                                                 \
+  }
+
+DEF_NAPI_COERCE_TO(bool, boolean);
+DEF_NAPI_COERCE_TO(number, number);
+DEF_NAPI_COERCE_TO(object, object);
+DEF_NAPI_COERCE_TO(string, string);
+
+napi_status napi_typeof(napi_env env, napi_value value,
+                        napi_valuetype* result) {
+  jerry_value_t jval = AS_JERRY_VALUE(value);
+  jerry_type_t type = jerry_value_get_type(jval);
+
+#define CM(jerry, napi) \
```

Comment:
What's the `CM` for?

Reply:
case-map，just a simple map abbreviation for type enums.

Reply:
How about using `SET` or `MAP` for the shorthand?